scanner.py

```diff
@@ -427,117 +516,594 @@ def start_touch(self, gcmd, touch_settings, verbose):
+        else:
+            has_increased_threshold_max = (
+                                True  # max limit is set, dont increase
+                            )
+
+        override = gcmd.get_int("OVERRIDE", 0)
+
+        confirmation_retries = gcmd.get_int(
+            "QUALIFY_SAMPLES", 5
+        )  # Number of consistent retries for a good candidate
+        repeat_attempts = gcmd.get_int(
+            "VERIFY_SAMPLES", 3
+        )  # Define repeat attempts for consistency check
+
+        # Define what qualifies as a "good" result
+        max_acceptable_retries = 3
+        max_acceptable_std_dev = vars["target"]
+
+        verbose = vars["verbose"]
+        csv_filename = (
+            f"/tmp/scanner_touch_scan_{time.strftime('%Y%m%d_%H%M%S')}.csv"
+            if verbose == 1
+            else None
+        )
+
+        # Prepare to track results
+        results = []
+
+        # Proceed with threshold scanning
+        self.check_temp(gcmd)
+        self.toolhead.wait_moves()
+
+        # Ensure XY homing
+        curtime = self.printer.get_reactor().monotonic()
+        kinematics = self.toolhead.get_kinematics()
+        kin_status = kinematics.get_status(curtime)
+        if "x" not in kin_status["homed_axes"] or "y" not in kin_status["homed_axes"]:
+            self.trigger_method = 0
+            raise gcmd.error("Must home X and Y axes first")
+
+        # Set initial scan values
+        self.previous_probe_success = 0
+        current_threshold = threshold_min
+        retries, attempts = 0, 0
+
+        try:
+            start_position = kin_status["axis_maximum"][2]
+            initial_position = self.toolhead.get_position()[:]
+            homing_position = initial_position[:]
+            initial_position[2], homing_position[2] = (
+                start_position,
+                kin_status["axis_minimum"][2],
+            )
+            max_accel = self.toolhead.get_status(curtime)["max_accel"]
+
+            # CSV setup if verbose
+            if verbose == 1:
+                with open(csv_filename, "w", newline="") as csvfile:
+                    csvwriter = csv.writer(csvfile)
+                    csvwriter.writerow(
+                        ["Sample Number", "Position (Z)", "Time (s)", "Threshold"]
+                    )
+
+            # Threshold scanning loop
+            while current_threshold <= threshold_max:
+                gcmd.respond_info(f"Testing Threshold value {current_threshold}...")
+                self.detect_threshold_z = current_threshold
+
+                touch_settings = TouchSettings(
+                    initial_position,
+                    homing_position,
+                    vars["accel"],
+                    vars["speed"],
+                    vars["retract_dist"],
+                    vars["retract_speed"],
+                    vars["num_samples"],
+                    vars["tolerance"],
+                    confirmation_retries,
```

Comment:
Why do we use `confirmation_retries` here when we drop the result here, if it above `max_acceptable_retries` https://github.com/Cartographer3D/cartographer-klipper/blob/2dfe5f0158384d850c4a00eca5594356520ce158/scanner.py#L777
Isn't it better to just use `confirmation_retries`, which is what the user potentially defined as well?

Reply:
We agreed to scale `max_acceptable_retries` off `confirmation_retries`. I added a scaling factor as a constant for now.